src/Interpreters/MySQL/InterpretersMySQLDDLQuery.cpp

```diff
@@ -117,6 +117,46 @@ static inline NamesAndTypesList getColumnsList(ASTExpressionList * columns_defin
     return columns_name_and_type;
 }
 
+static inline ColumnsDescription getColumnsDescription(NamesAndTypesList columns_name_and_type, ASTExpressionList * columns_define)
+{
+    if (columns_name_and_type.size() != columns_define->children.size())
+            throw Exception("Columns of different size provided.", ErrorCodes::BAD_ARGUMENTS);
+
+    ColumnsDescription columns_description;
+    std::queue<ColumnDescription> columns_queue;
```

Comment:
Is there a reason for using `std::queue` here that I don't understand, or could you use a fixed-size `std::array` instead? It's always good to avoid too many allocations.

Reply:
I suspect it is to be able to use an iterator in the for loop below? 

I think it would be better to change the loop below to loop over the index, and just create the ColumnDescription on the fly, making this loop unnecessary.

Reply:
Did it on the fly and removed queue completely.